share/wake/lib/system/job.wake

```diff
@@ -417,131 +533,142 @@ export def getJobStatus (job: Job): Status = match (getJobReport job)
-    Pair (Pass _) inFile =
-      def unlink f =
-        def fn _ = prim "unlink"
-        match (getenv "WAKEBOX_KEEP_IO_FILES")
-          Some "1" = Unit
-          _        = fn f
-      def outFile = replace `\.in\.json$` ".out.json" inFile
-      def json = parseJSONFile (Path outFile)
-      def _ = unlink inFile
-      match json
-        Fail f = Fail f
-        Pass content =
-          def _ = unlink outFile
-          def field name = match _ _
-             _ (Fail f) = Fail f
-             None (Pass fn) = Fail "{script} produced {outFile}, which is missing usage/{name}"
-             (Some x) (Pass fn) = Pass (fn x)
-          def usage = content // `usage`
-          def usageResult =
-            Pass (Usage _ _ _ _ _ _)
-            | field "status"   (usage // `status`   | getJInteger)
-            | field "runtime"  (usage // `runtime`  | getJDouble)
-            | field "cputime"  (usage // `cputime`  | getJDouble)
-            | field "membytes" (usage // `membytes` | getJInteger)
-            | field "inbytes"  (usage // `inbytes`  | getJInteger)
-            | field "outbytes" (usage // `outbytes` | getJInteger)
-          def getK exp = content // exp | getJArray | getOrElse Nil | mapPartial getJString
-          match usageResult
-            Fail f = Fail (makeError f)
-            Pass usage = Pass (RunnerOutput (getK `inputs`) (getK `outputs`) usage)
-  makeRunner "json-{script}" score pre post localRunner
+    def rawScript = plan.getJSONRunnerPlanRawScript
+    def extraArgs = plan.getJSONRunnerPlanExtraArgs
+    def extraEnv = plan.getJSONRunnerPlanExtraEnv
+    def score = plan.getJSONRunnerPlanScore
+    def estimate = plan.getJSONRunnerPlanEstimate
+
+    def script = which (simplify rawScript)
+    def ok = access script xOK
+    def pre = match _
+        Fail f = Pair (Fail f) ""
+        _ if ! ok = Pair (Fail (makeError "Runner {script} is not executable")) ""
+        Pass (RunnerInput label command visible environment directory stdin res prefix record) =
+            def Usage status runtime cputime membytes inbytes outbytes = record
+            def pmkdir m p = prim "mkdir"
+            def pwrite m p d = prim "write"
+            def json =
+                JObject
+                (
+                    "label" :-> JString label,
+                    "command" :-> command | map JString | JArray,
+                    "environment" :-> environment | map JString | JArray,
+                    "visible" :-> visible | map (_.getPathName.JString) | JArray,
+                    "directory" :-> JString directory,
+                    "stdin" :-> JString stdin,
+                    "resources" :-> res | map JString | JArray,
+                    "version" :-> JString version,
+                    "mount-ops" :-> JArray (JObject ("type" :-> JString "workspace", "destination" :-> JString ".", Nil), Nil),
+                    "usage"
+                    :-> JObject
+                    (
```

Comment:
Especially if we do go with the single-line format, can the formatter keep these all together?
```suggestion
                    "usage" :-> JObject (
```